text/0002-extensible-provisioning-templates.md

```diff
@@ -0,0 +1,118 @@
+-   Feature Name: Extensible provisioning templates
+-   Start Date: 2016-06-28
+-   RFC PR: (leave this empty)
+-   Issue: (leave this empty)
+
+# Summary
+
+[summary]: #summary
+
+This is a suggestion for refactoring Foreman's template generation mechanism is such a way that will enable
+different plugins to add plugin-specific code to the rendered template.
+
+# Motivation
+
+[motivation]: #motivation
+
+## Original use case
+
+As part of the effort to clean up Foreman's core from puppet related code,
+I have stumbled upon `Host::Managed#handle_ca` method. This method is responsible
+for dealing with certificate autosigning mechanism. The method is called from different parts
+in the code: from `UnattendedController` and from two orchestration concerns: `Orchestration::SSHProvision`
+and `Orchestration::Compute`. The motivation for this method is to keep the autosigning window open
+for as shortest period of time as possible.
+My suggestion is to solve this design flaw by using extensible templates in the following process:
+
+1.  Foreman will generate a certificate during the regular orchestration process and store it as part of
```

Comment:
Actually I am open to suggestions. I think it's the same problem with any secret that we want to pass to the host.